src/os_posix/os_mtx_cond.c

```diff
@@ -9,30 +9,22 @@
 #include "wt_internal.h"
 
 /*
- * __wt_cond_alloc --
- *	Allocate and initialize a condition variable.
+ * __wt_cond_init --
+ *	Initialize a condition variable.
  */
 int
-__wt_cond_alloc(WT_SESSION_IMPL *session, const char *name, WT_CONDVAR **condp)
+__wt_cond_init(WT_SESSION_IMPL *session, WT_CONDVAR *cond, const char *name)
 {
-	WT_CONDVAR *cond;
-	WT_DECL_RET;
-
-	WT_RET(__wt_calloc_one(session, &cond));
+	WT_UNUSED(session);
```

Comment:
I'd feel more comfortable if we cleared the content of `WT_CONDVAR` via memset in here.